tests/jsunit/event_test.js

```diff
@@ -392,3 +392,154 @@ function test_varBackard_runForward() {
   checkVariableValues(workspace, 'name1', 'type1', 'id1');
   eventTest_tearDown();
 }
+
+function test_events_filter() {
+  eventTest_setUpWithMockBlocks();
+  var block1 = workspace.newBlock('field_variable_test_block', '1');
+  var events = [
+    new Blockly.Events.BlockCreate(block1),
+    new Blockly.Events.BlockMove(block1),
+    new Blockly.Events.BlockChange(block1, 'field', 'VAR', 'item', 'item1'),
+    new Blockly.Events.Ui(block1, 'click')
+  ];
+  var filteredEvents = Blockly.Events.filter(events, true);
+  assertEquals(4, filteredEvents.length);  // no event should have been removed.
+  // test that the order hasn't changed
+  assertTrue(filteredEvents[0] instanceof Blockly.Events.BlockCreate);
+  assertTrue(filteredEvents[1] instanceof Blockly.Events.BlockMove);
+  assertTrue(filteredEvents[2] instanceof Blockly.Events.BlockChange);
+  assertTrue(filteredEvents[3] instanceof Blockly.Events.Ui);
+}
+
+function test_events_filterForward() {
+  eventTest_setUpWithMockBlocks();
+  var block1 = workspace.newBlock('field_variable_test_block', '1');
+  var events = [
+    new Blockly.Events.BlockCreate(block1),
+    new Blockly.Events.BlockMove(block1)
+  ];
+  block1.xy_ = new goog.math.Coordinate(1, 1);
+  events[1].recordNew();
+  events.push(new Blockly.Events.BlockMove(block1));
```

Comment:
I usually group helper functions at the bottom:

https://github.com/google/blockly/blob/master/tests/jsunit/connection_db_test.js#L272